dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java

```diff
@@ -83,14 +84,29 @@ public GHRepository getRepo(String repoName) throws IOException {
         return gitHubUtil.getRepo(repoName);
     }
 
-    public PagedSearchIterable<GHContent> findFilesWithImage(String image, String org) throws IOException {
+    public Optional<List<PagedSearchIterable<GHContent>>> findFilesWithImage(
+            String image,
+            Map<String, Boolean> orgsToIncludeOrExclude,
+            Integer gitApiSearchLimit) throws IOException {
         GHContentSearchBuilder search = gitHubUtil.startSearch();
         // Filename search appears to yield better / more results than language:Dockerfile
         // Root cause: linguist doesn't currently deal with prefixes of files:
         // https://github.com/github/linguist/issues/4566
         search.filename("Dockerfile");
-        if (org != null) {
-            search.user(org);
+        if (!orgsToIncludeOrExclude.isEmpty()) {
+            StringBuilder includeOrExcludeOrgsQuery = new StringBuilder();
+            for (Map.Entry<String, Boolean> org : orgsToIncludeOrExclude.entrySet()){
+                if (org.getKey() != null) {
+                    if (org.getValue()) {
+                        log.info("Including the org {} in the search.", org.getKey());
+                        includeOrExcludeOrgsQuery.append(String.format("user:%s ", org.getKey()));
+                    } else {
+                        log.info("Excluding the org {} from the search.", org.getKey());
+                        includeOrExcludeOrgsQuery.append(String.format("-org:%s ", org.getKey()));
+                    }
+                }
+            }
+            search.q(includeOrExcludeOrgsQuery.toString());
```

Comment:
Do we need to check to make sure we don't exceed any search term limits?

Reply:
Can you please point me to some documentation about search term limits?

Reply:
https://docs.github.com/en/rest/reference/search#limitations-on-query-length

Reply:
Thanks @justinharringa!
Do we need to add an additional check here or can we rely on the check that Github will do internally and return a "Validation failed" error if the limit is exceeded?

Reply:
I have not spotted anything in the github-api Java library that does this check but I'm not sure. I suspect that you'd likely get a failure from the actual GitHub API in this case.

Seems like it would be worth testing. A simple way to test would be to simply try and build a `main()` that adds 500 search terms of length beyond the limit and see what happens.

Reply:
I added 500 search terms to make sure that it exceeds the limit mentioned [here](https://docs.github.com/en/rest/reference/search#limitations-on-query-length) and got the following error 
`Caused by: org.kohsuke.github.HttpException: {"message":"Validation Failed","errors":[{"message":"The search is longer than 128 characters.","resource":"Search","field":"q","code":"invalid"}],"documentation_url":"https://docs.github.com/enterprise/3.1/v3/search/"}`

This validation is handled at the github api level, so maybe we do not need to add any more validations here.

Reply:
Ouch. That error shows that it only allows 128 characters. Is the thought that there would be some follow-on work to split out the search to individual org searches? It seems like it may not be hard to hit this limit for the overall search if we end up with something like this:
`filename:Dockerfile -org:somelonguser -org:someotherreallylonguser -org:anotherreallylonguser -org:ohnothisoneisfartoolongandwefailed`

Reply:
Running the search for every individual org will increase # of gh api calls a lot. But reaching a 128 chars limit would not be very hard like you mentioned.
CC @afalko

Reply:
I don't think we need to do a search for each individual org unless each of them are 90 long. I think we'll need do the following if the total length of orgs + the Dockerfile search is >120:
1) Sort orgs by length
2) Build list of smallest length orgs until you have as many as will fit under the 120 limit
3) Call search with those
4) Repeat step #2 until you don't have any orgs remaining. 
We'll need a function that calls search again with another set of orgs if

Reply:
I'll let you 2 make the determination on follow-up or not - would be good to check one of our known large orgs to get an idea of how impactful this is. Note that `github-api` does try and respect rate limiting in some cases with retries (although it's simple sleep where this does occur). I believe that this is one of those cases so that certainly helps with GH API calls.

Reply:
@justinharringa : we determined to proceed forward without the above and create a follow up change to either do the above or take another approach all together.